src/selector.rs

```diff
@@ -0,0 +1,24 @@
+//! Abstract Faiss ID selector
+use faiss_sys::*;
+
+/// Abstraction over IDSelectorRange and IDSelectorBatch
+#[derive(Debug)]
+pub struct IdSelector {
+    pub inner: *mut FaissIDSelector,
+}
+
+impl IdSelector {
+    /// Create new range selector
+    pub fn range(min: idx_t, max: idx_t) -> IdSelector {
+        let mut sel = FaissIDSelector_H {_unused: []};
+        let _ = faiss_IDSelectorRange_new(&mut &mut sel, min, max);
+        IdSelector { inner: &mut sel }
+    }
+
+    /// Create new batch selector
+    pub fn batch(n: i64, indices: &idx_t) -> IdSelector {
```

Comment:
These function arguments need to change too. `&idx_t` is a reference to a single element, whereas `&[Idx]` points to a contiguous slice of elements with a known length.

```suggestion
    pub fn batch(indices: &[Idx]) -> Result<Self> {
```